### PR TITLE
Update IMDB.cpp

### PR DESCRIPTION
**Pull request by fialky**
IMDB - adult movies scraping support on the base of "Enable adult movies scrapers" option. (part 2/2)

### CODE REVIEW

File: src/scrapers/movie/IMDB.cpp [hunk 1]
```diff
@@ -105,6 +105,7 @@ QString IMDB::defaultLanguageKey()
 void IMDB::search(QString searchStr)
 {
     QString encodedSearch = QUrl::toPercentEncoding(searchStr);
+    QString includeAdult = (Settings::instance()->showAdultScrapers()) ? "true" : "false";
```
Comment (bugwelle) [inline src/scrapers/movie/IMDB.cpp:108]: Better use a bool. There's not need to store it as a string. :-)

File: src/scrapers/movie/IMDB.cpp [hunk 2]
```diff
@@ -115,6 +116,15 @@ void IMDB::search(QString searchStr)
         new NetworkReplyWatcher(this, reply);
         connect(reply, &QNetworkReply::finished, this, &IMDB::onSearchIdFinished);
 
+    } else if (QVariant(includeAdult).toBool()) {
```
Comment (bugwelle) [inline src/scrapers/movie/IMDB.cpp:119]: Not necessary if `includeAdult` was a boolean. :-)

File: src/scrapers/movie/IMDB.cpp [hunk 3]
```diff
@@ -115,6 +116,15 @@ void IMDB::search(QString searchStr)
...
+        QUrl url = QUrl::fromEncoded(
+            QStringLiteral("https://www.imdb.com/search/title/?adult=include&view=simple&title=%1").arg(encodedSearch).toUtf8());
```
Comment (bugwelle) [inline src/scrapers/movie/IMDB.cpp:121]: Note: This URL will also include TV shows and other non-movies.

File: src/scrapers/movie/IMDB.cpp [hunk 4]
```diff
@@ -115,6 +116,15 @@ void IMDB::search(QString searchStr)
...
+        QNetworkRequest request(url);
+        request.setRawHeader("Accept-Language", "en"); // todo: add language dropdown in settings
```
Comment (bugwelle) [inline src/scrapers/movie/IMDB.cpp:123]: Btw: The results of IMDb are based on the users location and not this header. Setting this header is not necessary. That's something that I learned just recently... :-/

File: src/scrapers/movie/IMDB.cpp [hunk 5]
```diff
@@ -197,6 +228,24 @@ void IMDB::onSearchIdFinished()
     emit searchDone(results, {});
 }
 
+QVector<ScraperSearchResult> IMDB::parseSearchAdult(QString html)
+{
+    QVector<ScraperSearchResult> results;
+
+    QRegExp rx("<a href=\"/title/([t]*[\\d]+)/[^\"]*\"\\n>([^<]*)</a>\\n.*(?: \\(I+\\) |>)\\(([0-9]*).*\\)");
```
Comment (bugwelle) [inline src/scrapers/movie/IMDB.cpp:235]:
```suggestion
    QRegExp rx("<a href=\"/title/(tt[\\d]+)/[^\"]*\"\\n>([^<]*)</a>\\n.*(?: \\(I+\\) |>)\\(([0-9]*).*\\)");
```

IMDb IDs always start with `tt` :-)

File: src/scrapers/movie/IMDB.cpp (hunk 1 above)
Comment (fialky) [reply src/scrapers/movie/IMDB.cpp]: Thank you for your bool suggestion. Tested and included.

File: src/scrapers/movie/IMDB.cpp (hunk 3 above)
Comment (fialky) [reply src/scrapers/movie/IMDB.cpp:121]: Thank you for your bool suggestion. Tested and included.

File: src/scrapers/movie/IMDB.cpp (hunk 2 above)
Comment (fialky) [reply src/scrapers/movie/IMDB.cpp]: Thank you for your bool suggestion. Tested and included (ce0587b).

File: src/scrapers/movie/IMDB.cpp (hunk 5 above)
Comment (fialky) [reply src/scrapers/movie/IMDB.cpp]: The suggestion is tested and works well after recompilation (4082e89). Thank you!

File: src/scrapers/movie/IMDB.cpp (hunk 3 above)
Comment (fialky) [reply src/scrapers/movie/IMDB.cpp:121]: Do you think that the addition of "&title_type=feature,documentary,short,video" into the adult URL is better than nothing :-)?

Comment (bugwelle) [reply src/scrapers/movie/IMDB.cpp:121]: Yes, that should work, e.g. for The Simpsons: https://www.imdb.com/search/title/?adult=include&view=simple&title=Simpsons&title_type=feature,documentary,short,video

The series isn't listed. :)

Btw:

File: src/scrapers/movie/IMDB.cpp [hunk 6]
```diff
@@ -105,7 +105,7 @@ QString IMDB::defaultLanguageKey()
 void IMDB::search(QString searchStr)
 {
     QString encodedSearch = QUrl::toPercentEncoding(searchStr);
-    QString includeAdult = (Settings::instance()->showAdultScrapers()) ? "true" : "false";
+    bool includeAdult = (Settings::instance()->showAdultScrapers()) ? "true" : "false";
```
Comment (bugwelle) [inline src/scrapers/movie/IMDB.cpp:108]:
```suggestion
    bool includeAdult = Settings::instance()->showAdultScrapers();
```

File: src/scrapers/movie/IMDB.cpp (hunk 3 above)
Comment (fialky) [reply src/scrapers/movie/IMDB.cpp]: TV shows and other non-movies filtered out (baded46).

**Comment by fialky** [comment]
Sometimes `"datePublished":*` or `<h4 class="inline">Release Date:</h4>*` are not present. Better than nothing, we can scrape the year from `<title>*` (1aef8a9).

**Comment by bugwelle** [comment]
Took your changes and refactored the code a bit. I close this issue in favor of #992

**Pull request closed without merging**